plugins/module_utils/openssh/utils.py

```diff
@@ -33,3 +66,309 @@ def parse_openssh_version(version_string):
         version = None
 
     return version
+
+
+# See https://datatracker.ietf.org/doc/html/rfc4251#section-5 for SSH data types
+class OpensshParser(object):
+    """Parser for OpenSSH encoded objects"""
+    BOOLEAN_OFFSET = 1
+    UINT32_OFFSET = 4
+    UINT64_OFFSET = 8
+
+    def __init__(self, data):
+        if not isinstance(data, (bytes, bytearray)):
+            raise TypeError("Data must be bytes-like not %s" % type(data))
+
+        self._data = memoryview(data) if PY3 else data
+        self._pos = 0
+
+    def boolean(self):
+        self._check_position(self.BOOLEAN_OFFSET)
+
+        next_pos = self._pos + self.BOOLEAN_OFFSET
```

Comment:
How about making `_check_position` return `next_pos`, so you can simply write:
```suggestion
        next_pos = self._check_position(self.BOOLEAN_OFFSET)
```
That would simplify quite a few of the next methods.